### DISCUSSION TITLE
Are there some ways to make Quest and Rewards available on VR?

### DISCUSSION THREAD

**Discussion by giaddah** [spatialsys/spatial-unity-sdk#9] [Help] [answered]
Hello everyone,
I have a question regarding the use of Quests and Rewards on all the Spatial platforms. I'm currently using Meta Quest2 and I'm working on create Quests on my project. Testing the app on web, they normally show up but I need them to work also on VR. I'm beginning to understand they are not supported on VR yet, am I right? Is there some way I can let this happen on my VR project?

**Comment by andrepucas** [comment, marked as answer]
The system itself is supported, the biggest limitation is the UI display. While in Web we get the bottom left progress tracker (bottom center in the mobile app). On VR, because there's no persistent UI that follows the player camera, the only feedback the user gets is a message that appears in the middle of the screen, upon quest task completed. Markers also work, if I remember correctly. I think that while VR doesn't get a proper UI revamp, the issue will persist. 

You can try to take things a bit outside the box tho (it's a lot more work tho) and have one or more physical places where you manually display quest progress. It could be a fun and immersive way to do quests on VR

**Reply by giaddah** [reply]
Thank you for answering. I also thought about creating a UI display from scratch, in order to view it on VR. Something like a giant static panel that marks all the progresses of the tasks. But since I'm kinda new to Unity and Creator Toolkit I have to figure it out how to do this. Thanks again tho!

**Reply by andrepucas** [reply]
In the past I've also tried to have a World Space Canvas following the player's position + a forward offset, so that it was always in front of the player and following his rotation, on Update, but it ended up looking a bit janky ahah

**Comment by giaddah** [comment]
I’m not so familiar with scripting but in general I have issues on putting
everything that has Event System in it, such as buttons or canvas, because
they are not supported on Spatial

**Reply by AlanSpatial** [reply]
You can tag an event system as `editor only` to use it for testing in editor, but exclude it from builds.

![image](https://github.com/spatialsys/spatial-unity-sdk/assets/100000804/ee0a6c8b-8232-4097-a7cf-16fe692c3bca)
